channeldb/forwarding_log_test.go

```diff
@@ -360,3 +361,213 @@ func TestForwardingLogStoreEvent(t *testing.T) {
 		}
 	}
 }
+
+// TestForwardingLogQueryIncomingChanIDs tests that querying the forwarding log
+// using only incoming channel IDs returns the correct subset of events.
+func TestForwardingLogQueryIncomingChanIDs(t *testing.T) {
+	t.Parallel()
+
+	// Set up a test database.
+	db, err := MakeTestDB(t)
+	require.NoError(t, err, "unable to make test db")
+
+	log := ForwardingLog{
+		db: db,
+	}
+
+	initialTime := time.Unix(1234, 0)
+	endTime := time.Unix(1234, 0)
+
+	// Create 10 random events with varying incoming ChanIDs.
+	numEvents := 10
+	events := make([]ForwardingEvent, numEvents)
+	incomingChanIDs := []lnwire.ShortChannelID{
+		lnwire.NewShortChanIDFromInt(2001),
+		lnwire.NewShortChanIDFromInt(2002),
+		lnwire.NewShortChanIDFromInt(2003),
+	}
+
+	for i := 0; i < numEvents; i++ {
+		events[i] = ForwardingEvent{
+			Timestamp:      endTime,
+			IncomingChanID: incomingChanIDs[i%len(incomingChanIDs)],
+			AmtIn:          lnwire.MilliSatoshi(rand.Int63()),
+			AmtOut:         lnwire.MilliSatoshi(rand.Int63()),
+		}
+		endTime = endTime.Add(time.Minute * 10)
+	}
+
+	// Add events to the database.
+	require.NoError(t, log.AddForwardingEvents(events),
+		"unable to add events")
+
+	// Query with multiple incoming channel IDs.
+	eventQuery := ForwardingEventQuery{
+		StartTime: initialTime,
+		EndTime:   endTime,
+		IncomingChanIDs: fn.NewSet(incomingChanIDs[0].ToUint64(),
```

Comment:
nit: format these `fn.NewSet()` calls like other multi-line function calls.

Technically, the `require.NoError()` calls also fall under that rule, since they don't use string formatting templates (e.g. the function name doesn't end in an `f` to indicate formatting).